src/wasm.h

```diff
@@ -1365,50 +515,18 @@ class Function {
 
   Function() : result(none) {}
 
-  size_t getNumParams() {
-    return params.size();
-  }
-  size_t getNumVars() {
-    return vars.size();
-  }
-  size_t getNumLocals() {
-    return params.size() + vars.size();
-  }
+  size_t getNumParams();
+  size_t getNumVars();
+  size_t getNumLocals();
 
-  bool isParam(Index index) {
-    return index < params.size();
-  }
-  bool isVar(Index index) {
-    return index >= params.size();
-  }
+  bool isParam(Index index);
+  bool isVar(Index index);
 
-  Name getLocalName(Index index) {
-    assert(index < localNames.size() && localNames[index].is());
-    return localNames[index];
-  }
-  Name tryLocalName(Index index) {
-    if (index < localNames.size() && localNames[index].is()) {
-      return localNames[index];
-    }
-    // this is an unnamed local
-    return Name();
-  }
-  Index getLocalIndex(Name name) {
-    assert(localIndices.count(name) > 0);
-    return localIndices[name];
-  }
-  Index getVarIndexBase() {
-    return params.size();
-  }
-  WasmType getLocalType(Index index) {
-    if (isParam(index)) {
-      return params[index];
-    } else if (isVar(index)) {
-      return vars[index - getVarIndexBase()];
-    } else {
-      WASM_UNREACHABLE();
-    }
-  }
+  Name getLocalName(Index index);
+  Name tryLocalName(Index index);
```

Comment:
I actually think `tryLocalName` is pretty clear - it tries to get it instead of just assuming it's there. Just me? :)

I'm ok with @dschuff's suggestion too.